### DISCUSSION TITLE
A mechanism to protect things like API keys and passwords for interfacing with other microservices/backends

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4364] [Feedback]
A user says:
>Wanted to let you know that I finally set aside some time this weekend to try out Dark! I went through all of the pages on the "docs" site, and setup the sample SPA react app.
>
>The primary feature that would keep me from using this on something semi-work related is the absence of a mechanism to protect things like API keys and passwords for interfacing with other microservices/backends. However, everything else has been absolutely wonderful!
>
>The developer ergonomics are so nice, and I hope some ideas to make "devtools" around the canvas will emerge.

**Comment by hydrosquall** [comment]
Hi, this was me! I found this useful when developing https://covidcommitment.org/ , which executes lambda functions that wrap some Python logic in addition to using Dark. I ended up moving the parts of my code that required using API keys into the Python layer instead, but that work could easily have been done in Dark instead.
